docs/advanced/customization-guide.md

```diff
@@ -16,8 +15,1007 @@ published: false
 
 ---
 
-***WORK IN PROGRESS.***
+## Overview
+
+NFD provides multiple extension points for vendor and application specific
+labeling:
+
+- [NodeFeatureRule](#nodefeaturerule-custom-resource) objects provide a way to
+  deploy custom labeling rules via the Kubernetes API
+- [*local* feature source](#local-feature-source) of nfd-worker creates
+  labels by executing hooks and reading files
+- [*custom* feature source](#custom-feature-source) of nfd-worker creates
+  labels based on user-specified rules
+
+## NodeFeatureRule custom resource
+
+NodeFeatureRule objects provide an easy way to create vendor or application
+specific labels. It uses a flexible rule-based mechanism for creating labels
+based on node feature.
+
+### A NodeFeatureRule example
+
+Consider the following referential example:
+
+```yaml
+apiVersion: nfd.k8s-sigs.io/v1alpha1
+kind: NodeFeatureRule
+metadata:
+  name: my-rule-object
+spec:
+  rules:
+    - name: "my sgx feature rule"
+      labels:
+        "my-sgx-feature": "true"
```

Comment:
nit: even if this is just an example, to make it work in all cases add (or change `cpu.cpuid`):
```
     - feature: cpu.sgx
        matchExpressions:
          enabled: {op: IsTrue}
```

Reply:
Yeah, I know. The idea behind the example was to have a rule with more than one matcher. Maybe I'll change this to something completely different like `cpu.pststate.turbo` and `kernel.config.PREEMPT` or smth 🤔

Reply:
Changed this to
```
      matchFeatures:
        - feature: kernel.loadedmodule
          matchExpressions:
            dummy: {op: Exists}
        - feature: kernel.config
          matchExpressions:
            X86: {op: In, value: ["y"]}
```